src/site/_collections/algolia.js

```diff
@@ -58,10 +64,12 @@ module.exports = (collection) => {
   // For now, hard-code language to English.
   const lang = 'en';
 
-  const authorsCollection = authorsCollectionFn();
+  const authorsCollection = authorsCollectionFn(collection);
+  const newslettersCollection = newslettersCollectionFn(collection);
+  const tagsCollection = tagsCollectionFn(collection);
```

Comment:
I'm wondering if we can avoid having to recompute these collections.

You can define a javascript function in the _data directory and pass collections to it as an argument. So in algolia.njk you could do something like:

```
{{ helpers.algolia(collections.all, collections.authors, collections.newsletters, collections.tags) }}
```

Where helpers.algolia is defined as:

```
// helpers.js

const algolia = (all, authors, newsletter, tags) => {
  ...
  // return some json
}
module.exports = {algolia};
```

Reply:
So other than the newsletter, author and tags does not get recomputed. In earlier feature I added the authors and tags gets stored and returned instead of it being we recomputed.

I could do the same for the newsletter, but let me know what you think.

Reply:
I think my preference is for these dependencies to be passed in as arguments. It feels a bit cleaner and more testable since we could mock them. (side note, let's add some tests as well).

Memoizing the collections works, but it feels like we're working around the system. I keep thinking that eleventy collections should be as simple as possible because they have such a funky lifecycle, and if we can pass them to helper functions that live in _data then we might end up with simpler, more testable code.

Reply:
I set the "helper" as a filter in `_filter`, it made more sense to me there, though the logic should behave the same in any folder. It just didn't make sense in the data folder as those should be just, well, data objects.